treeherder/perf/management/commands/migrate_perf_signature_subtest_schema.py

```diff
@@ -0,0 +1,42 @@
+from optparse import make_option
+
+from django.core.management.base import BaseCommand
+
+from treeherder.model.models import (Datasource,
+                                     Repository)
+from treeherder.perf.models import PerformanceSignature
+
+
+class Command(BaseCommand):
+    help = "make subtest store as foreign key"
+    option_list = BaseCommand.option_list + (
+        make_option('--project',
+                    action='append',
+                    dest='project',
+                    help='Filter deletion to particular project(s)',
+                    type='string'),
+    )
+
+    def handle(self, *args, **options):
+        if options['project']:
+            projects = options['project']
+        else:
+            projects = Datasource.objects.values_list('project', flat=True)
+
+        for project in projects:
+            print project
+            try:
+                repository = Repository.objects.get(name=project)
+            except:
+                continue
+            signatures = PerformanceSignature.objects.filter(
+                    repository=repository
+                )
+            for signature in signatures:
+                if signature.extra_properties.get('subtest_signatures'):
+                    subtest_signatures = signature.extra_properties.pop('subtest_signatures')
+                    signature.save()
+                    subtests = PerformanceSignature.objects.filter(
+                        signature_hash__in=subtest_signatures)
+                    for subtest in subtests:
+                        subtest.parent_singature = signature
```

Comment:
There's a typo here (`singature`). I theorize that you haven't yet tested this. :)

Anyway, the idea is right, but we need to be a bit careful here, as there might be multiple summary signatures with different child signatures (for example: DAMP). I think we'd probably be safe if we just migrated summaries and signatures that have been changed in, say, the last week. 

I suspect doing the migration properly might depend on having database access -- can you make sure that this works for a small set of performance data (say, by using ingest push with an old copy of treeherder: http://treeherder.readthedocs.org/installation.html#ingesting-a-single-push-at-a-time). If it does, then I can take care of the rest.